drivers/sensor/r502a/Kconfig

```diff
@@ -0,0 +1,62 @@
+# R502A hzgrow Fingerprint sensor Configuration options
+
+# Copyright (c) 2021 Linumiz
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig R502A
```

Comment:
make this `GROW_R502A`

Reply:
But, If the compatible name changes to hzgrow,r502a then it is better to have r502a as driver and config name right?

Reply:
@galak  If you think otherwise, I am ready to push the changes you asked for. Do I have to rename driver file names and kconfig entries to grow_r502a and GROW_R502A or keep the current changes as r502a alone?

Reply:
@galak Renamed as per your comments.

Reply:
@galak ping, kindly re-review the changes and help me mainlining this driver.